src/program.ts

```diff
@@ -74,9 +76,45 @@ function setupExitWatchdog(serverList: ServerList) {
 
 program.parse(process.argv);
 
-async function startSSEServer(port: number, serverList: ServerList) {
-  const sessions = new Map<string, SSEServerTransport>();
+async function startHttpServer(port: number, serverList: ServerList) {
+  const sseSessions = new Map<string, SSEServerTransport>();
+  const streamableSessions = new Map<string, StreamableHTTPServerTransport>();
   const httpServer = http.createServer(async (req, res) => {
+    const isStreamableHttp = req.url?.startsWith('/mcp');
```

Comment:
Please please kick this code into a separate file!

Reply:
done, and reorganised it